crates/primitives/derive/src/storable.rs

```diff
@@ -0,0 +1,140 @@
+// Copyright 2018-2022 Parity Technologies (UK) Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use proc_macro2::TokenStream as TokenStream2;
+use quote::{
+    quote,
+    quote_spanned,
+};
+use syn::spanned::Spanned;
+
+/// `Storable` derive implementation for `struct` types.
+fn storable_struct_derive(s: &synstructure::Structure) -> TokenStream2 {
+    assert_eq!(s.variants().len(), 1, "can only operate on structs");
+    let variant: &synstructure::VariantInfo = &s.variants()[0];
+    let decode_body = variant.construct(|field, _index| {
+        let ty = &field.ty;
+        let span = ty.span();
+        quote_spanned!(span =>
+            <#ty as ::ink_primitives::traits::Storable>::decode(__input)?
+        )
+    });
+    let encode_body = variant.each(|binding| {
+        let span = binding.ast().ty.span();
```

Comment:
nit: Can we get the type name in here so it's something like: `__balances_binding` instead?

Reply:
Didn't get you=)

Reply:
When reading the generated code this is `__binding0`, `__binding1`, etc. So just for the dev's sake I figured it would be nicer to make it more readable. Again, it's a nitpick so you don't have to change anything if you don't see value in it

Reply:
I got your idea. I tried to do that, but it means removing the usage of `synstructure::Structure` and implementing everything by myself. I prefer to avoid it=D